tests/unit/modules/test_storage_policies.py

```diff
@@ -0,0 +1,325 @@
+                                    ],
+                                )
+                            ],
+                        ),
+                        Mock(
+                            dynamicProperty=[],
+                            id=Mock(dynamicProperty=[], namespace="VSAN", id="iopsLimit"),
+                            constraint=[
+                                Mock(
+                                    dynamicProperty=[],
+                                    propertyInstance=[
+                                        Mock(dynamicProperty=[], id="iopsLimit", value=0)
+                                    ],
+                                )
+                            ],
+                        ),
+                        Mock(
+                            dynamicProperty=[],
+                            id=Mock(dynamicProperty=[], namespace="VSAN", id="cacheReservation"),
+                            constraint=[
+                                Mock(
+                                    dynamicProperty=[],
+                                    propertyInstance=[
+                                        Mock(dynamicProperty=[], id="cacheReservation", value=0)
+                                    ],
+                                )
+                            ],
+                        ),
+                        Mock(
+                            dynamicProperty=[],
+                            id=Mock(
+                                dynamicProperty=[], namespace="VSAN", id="proportionalCapacity"
+                            ),
+                            constraint=[
+                                Mock(
+                                    dynamicProperty=[],
+                                    propertyInstance=[
+                                        Mock(
+                                            dynamicProperty=[], id="proportionalCapacity", value=70
+                                        )
+                                    ],
+                                )
+                            ],
+                        ),
+                    ],
+                )
+            ],
+        ),
+        generationId=0,
+        isDefault=False,
+        spec=pbm.profile.CapabilityBasedProfile,
+    )
+
+
+@pytest.fixture
+def fake_service_instance(request):
+    # This fixture should be used for all unit tests where a service instance
+    # is needed. It will test both scenarios where the service instance is
+    # provided, or not.
+
+    with patch("saltext.vmware.utils.connect.get_service_instance", autospec=True) as fake_get_si:
+        fake_get_si.side_effect = Exception(
+            "get_service instance was unexpectedly called in a test"
+        )
+        fake_get_si.return_value._stub = MagicMock()
+        fake_get_si.return_value._stub.host = "localhost"
+        fake_get_si.return_value._stub.cookie = '"session_cookie"'
+        yield fake_get_si, fake_get_si.return_value
```

Comment:
**question** isn't fake_service_instance in the conftest.py fixture?

Reply:
It's almost the same, but I need this lines for 'pyVmomi.pbm.ServiceInstance':
```
fake_get_si.return_value._stub = MagicMock()
fake_get_si.return_value._stub.host = "localhost"
fake_get_si.return_value._stub.cookie = '"session_cookie"'
```

Reply:
done

Reply:
That makes sense - I don't think it's worth blocking this PR over, but I'll go ahead and create an issue to track this. You can extend fixtures like:

```
@pytest.fixture
def custom_si(fake_service_instance):
    ...  # whatever changes you need to make
    yield fake_service_instance   # typically yield, unless the result is static, like `return 42` or a dictionary or something
```

I'll go ahead and create an issue from this to track efforts